base/mapiterator.jl

```diff
@@ -0,0 +1,249 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+#-------------------------------------------------------------------------------
+# MapItr
+#
+# Simple synchronous map iterator.
+#
+#   collect(MapItr(f, c...)) == map(f, c...)
+#
+#-------------------------------------------------------------------------------
+
+import Base: start, done, next
+
+type MapItr
+    f::Union{Function,Type}
+    arg_itr
+end
+
+MapItr(f, c...) = MapItr(f, zip(c...))
+
+
+start(itr::MapItr) = start(itr.arg_itr)
+
+done(itr::MapItr, state) = done(itr.arg_itr, state)
+
+function next(itr::MapItr, state)
+    args, state = next(itr.arg_itr, state)
+    itr.f(args...), state
+end
+
+
+
+#-------------------------------------------------------------------------------
+# AsyncMapItr
+#
+# Asynchronous map iterator.
+# Creates a @sync block (between start() and done().
+# Executes mapped function calls using @asyc.
+# Function results are asynchronously stored in "results" collection
+# (iterator returns nothing).
+#
+#   for task in AsyncMapItr(f, results, c...) end
+#
+#   ==
+#
+#   map!(f, results, c...)
+#
+# The maximum number of concurrent @async tasks can be set with the
+# "ntasks" option. e.g.
+#
+#    AsyncMapItr(f, results, c...; ntasks=10)
+#
+#-------------------------------------------------------------------------------
+
+
+type AsyncMapItr
+    f
+    results
+    arg_enum::Enumerate
+    ntasks::Int
+end
+
+function AsyncMapItr(f, results, c...; ntasks=nothing)
+    if ntasks == nothing
```

Comment:
There are a whole stack of calls that pass `nothing` as the default.
Don't want to have to change multiple places when changing the default.